# Part 6 working with data

In [1]:
import pandas as pd
import numpy as np

/usr/lib/python3/dist-packages/pytz/__init__.py:31: SyntaxWarning: invalid escape sequence '\s'
  match = re.match("^#\s*version\s*([0-9a-z]*)\s*$", line)


## issues with the chicago_food_inspections data
* whitespaces surrounding the name
* inconsistency in names: uppercase and lowercase

In [7]:
inspections = pd.read_csv('/home/diego/Documents/Data/chicago_food_inspections.csv')
inspections['Name'].values

array([' MARRIOT MARQUIS CHICAGO   ', ' JETS PIZZA ', '   ROOM 1520 ',
       ..., ' Cafe 608 ', "  mr.daniel's ", '   TEMPO CAFE '],
      dtype=object)

In [31]:
# Modifying the column, first applying the method, and then replacing the column. For the two columns
for column in inspections.columns:
    inspections[column] = inspections[column].str.strip()

array(['Risk 1 (High)', 'Risk 2 (Medium)', 'Risk 3 (Low)', ...,
       'Risk 1 (High)', 'Risk 1 (High)', 'Risk 1 (High)'], dtype=object)

In [32]:
inspections

Name             Risk
0            MARRIOT MARQUIS CHICAGO    Risk 1 (High)
1                         JETS PIZZA  Risk 2 (Medium)
2                          ROOM 1520     Risk 3 (Low)
3            MARRIOT MARQUIS CHICAGO    Risk 1 (High)
4                         CHARTWELLS    Risk 1 (High)
...                              ...              ...
153805                     WOLCOTT'S    Risk 1 (High)
153806  DUNKIN DONUTS/BASKIN-ROBBINS  Risk 2 (Medium)
153807                      Cafe 608    Risk 1 (High)
153808                   mr.daniel's    Risk 1 (High)
153809                    TEMPO CAFE    Risk 1 (High)

[153810 rows x 2 columns]

## How can we know all different values a column can contain?

With the `unique()` method

In [33]:
# In the Risk column we have 5 different possible values in all the column
inspections.Risk.unique()

array(['Risk 1 (High)', 'Risk 2 (Medium)', 'Risk 3 (Low)', 'All', nan],
      dtype=object)

I just will remove the nan values

From original 153,810 rows, dropping nan we have 153,744 rows

In [38]:
inspections = inspections.dropna(subset=['Risk'])
inspections

Name             Risk
0            MARRIOT MARQUIS CHICAGO    Risk 1 (High)
1                         JETS PIZZA  Risk 2 (Medium)
2                          ROOM 1520     Risk 3 (Low)
3            MARRIOT MARQUIS CHICAGO    Risk 1 (High)
4                         CHARTWELLS    Risk 1 (High)
...                              ...              ...
153805                     WOLCOTT'S    Risk 1 (High)
153806  DUNKIN DONUTS/BASKIN-ROBBINS  Risk 2 (Medium)
153807                      Cafe 608    Risk 1 (High)
153808                   mr.daniel's    Risk 1 (High)
153809                    TEMPO CAFE    Risk 1 (High)

[153744 rows x 2 columns]

Let's check now the values that Risk column contains

In [39]:
inspections.Risk.unique()

array(['Risk 1 (High)', 'Risk 2 (Medium)', 'Risk 3 (Low)', 'All'],
      dtype=object)

Now setting 'All' to 'Risk 4 (Extreme)' to keep consistency in the format

In [44]:
# inspections.Risk = inspections.Risk.str.replace('All', 'Risk 4 (Extreme)')  not proper
inspections = inspections.replace('All', 'Risk 4 (Extreme)')  # this is a method of the df

/tmp/ipykernel_17304/605926085.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  inspections.Risk = inspections.Risk.str.replace('All', 'Risk 4 (Extreme)')


In [47]:
inspections.Risk.unique()

array(['Risk 1 (High)', 'Risk 2 (Medium)', 'Risk 3 (Low)',
       'Risk 4 (Extreme)'], dtype=object)